### ISSUE TITLE
Call dynamic method statically allowed

### ISSUE THREAD

**Issue by simPod** [phpstan/phpstan-strict-rules#102]
This is not allowed https://phpstan.org/r/21925d86-0e11-4c0f-a923-4a828e476ec3:

> Dynamic call to static method World::say().

```php
$this->say('Hello, ' . $hi);
...
public static function say(string $hi): void
```

while this is:

```php
self::say('Hello, ' . $hi);
...
public function say(string $hi): void
```

https://phpstan.org/r/c49e4ff4-5537-4ad1-8625-7bad0002157f

Shouldn't it be also reported as an issue?

**Comment by JanTvrdik** [comment]
`self::say` is not a static call. Neither is `parent::say` or `Foo::say`.

**Comment by simPod** [comment]
oh ok, thanks for explanation!
